### PR TITLE
Core/Misc: Added Loading.Locales to enable/disable the load of locales

### PR DESCRIPTION
**Pull request by BertiRean**
**Changes proposed:**

- Added a Bool Config for enable/disable the load of locales for develop enviroments.  It could save ~1GB or 800 MB of RAM
-  
-  

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**

(Does it build, tested in-game, etc.)
- Build pass
- The RAM consumed by worldserver decrease in 800MB or 1 GB approx

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by BertiRean** [comment]
@Aokromes Which way?

**Comment by Aokromes** [comment]
disable load of ruru, ptbr, znch, locales and load for example eses esmx.

**Comment by BertiRean** [comment]
I just want use english per default and not load the rest of them

**Comment by Aokromes** [comment]
> I just want use english per default and not load the rest of them

yes, but maybe someone wants to use english and own language.

**Comment by Nyr97** [comment]
maybe we can add a string instead to force load of all if empty, and if not, load input ones, esES, enGB/enUS, etc.

**Comment by Aokromes** [comment]
> maybe we can add a string instead to force load of all if empty, and if not, load input ones, esES, enGB/enUS, etc.

better bitmask xd

**Comment by Nyr97** [comment]
or using EnumeratedStrings index for Locales: https://old.wow.tools/dbc/?dbc=enumeratedstring&build=2.5.1.39640#page=1&colFilter[4]=3

**Comment by BertiRean** [comment]
> > maybe we can add a string instead to force load of all if empty, and if not, load input ones, esES, enGB/enUS, etc.
> 
> better bitmask xd

I like this more. A BitMask will be less painful

**Comment by BertiRean** [comment]
@Shauren BitMask or just a generic disable for all is a better option ?

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -41,6 +41,7 @@
 #    PACKET SPOOF PROTECTION SETTINGS
 #    METRIC SETTINGS
 #    PVP SETTINGS
+#	 LOADING SETTINGS
```
Comment (jackpoz) [inline src/server/worldserver/worldserver.conf.dist:44]:
```suggestion
#    LOADING SETTINGS
```

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -4424,3 +4425,13 @@ Pvp.FactionBalance.Pct20 = 0.8
...
 #
 ###################################################################################################
+
+###################################################################################################
+# LOADING SETTINGS
+#    Loading.Locales
+#        Description: Enable/Disable  the load of Locales for Creatures, gossips, quests and gameobjects.
+#        Default:     0 - (Disabled)
+#                     1 - (Enabled)
+Load.Locales = 1
```
Comment (Aokromes) [inline src/server/worldserver/worldserver.conf.dist:4435]: missing white line before and after.

File: src/server/worldserver/worldserver.conf.dist [hunk 3]
```diff
@@ -4424,3 +4425,13 @@ Pvp.FactionBalance.Pct20 = 0.8
 
 #
 ###################################################################################################
+
+###################################################################################################
+# LOADING SETTINGS
```
Comment (Aokromes) [inline src/server/worldserver/worldserver.conf.dist:4430]: missing white line before.

**Review by mdX7 — APPROVED** [review]

File: src/server/worldserver/worldserver.conf.dist [hunk 4]
```diff
@@ -4424,3 +4425,18 @@ Pvp.FactionBalance.Pct20 = 0.8
 
 #
 ###################################################################################################
+
+###################################################################################################
+# LOAD SETTINGS
+#
+# These settings control content loading
+#
```
Comment (Aokromes) [inline src/server/worldserver/worldserver.conf.dist:4433]: remove this white line.

Comment (mdX7) [reply src/server/worldserver/worldserver.conf.dist:4433]: no, keeping it for logical separation from section description. Same way has been used for PACKET SPOOF PROTECTION SETTINGS and METRIC SETTINGS a bit further up

**Comment by mdX7** [comment]
comparison:
```
36s startup time 4gb memory (locales disabled)
56s startup time 6.9gb memory (locales enabled) 
```

**Pull request merged**